subsystems/networking/networkmanager_linux.go

```diff
@@ -879,6 +885,25 @@ func (n *Networking) mainLoop(ctx context.Context) {
 				if n.Config().TurnOnHotspotIfWifiHasNoInternet.Get() {
 					lastConnectivity = n.connState.getLastOnline()
 				}
+			} else if !connectOk && userInputReceived && userInputSSID != "" {
+				// Special case: received AP credentials and exited provisioning to try it, but it's unsuccessful.
+				// Most commonly a wrong password. In roaming mode, we can get here too if we're connected but offline.
+				// Note: the network tests of `allGood` are updated in the background loop, so we may not get here
+				// in the same iteration as when user input was actually received, but rather a future iteration -
+				// `userInputReceived` and `userInputSSID` will have already been wiped.
+
+				// LockingNetwork is called with these same args in portalSave and tryCandidates; so should exist.
+				nw := n.netState.LockingNetwork(n.Config().HotspotInterface, userInputSSID)
```

Comment:
You acquire a locking network, but never actually lock it before reading the fields in it. As noted on the function itself, you should lock the mutex of the returned network object.

Alternately, if you want read-only values (which seems to be the case here), just use .Network() instead of .LockingNetwork()